ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobMetadata.java

```diff
@@ -505,6 +505,9 @@ public boolean equals(Object o) {
     if (!(o instanceof CloudBlobMetadata)) {
       return false;
     }
+    if (o == this) {
+      return true;
+    }
```

Comment:
this should return false.

Reply:
hmm if we are comparing an object to itself, the equals() should be true right ?
o.equals(o) == true ?

Reply:
oh, missed that